tests/ConnectDisconnectTest.test.nut

```diff
@@ -0,0 +1,176 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+@include "./ConnectionManager.lib.nut"
+
+class ConnectDisconnectTest extends ImpTestCase {
+
+  cm = null;
+  static TIMEOUT_1 = 5;
+  static TIMEOUT_2 = 10;
+  static TIMEOUT_3 = 15;
+  static TIMEOUT_4 = 20;
+
+  function setUp() {
+    this.info("running setUp");
+    cm = ConnectionManager();
+  }
+
+  /*
+  * disconnects and connects device using CM
+  *
+  */
+  function testSingleConnectDisconnectCMAsync() {
+    this.assertTrue(server.isconnected(), "should be connected!");
```

Comment:
there is no need to use this, this will make the code a little bit more compact and easy to read